aya/src/obj/mod.rs

```diff
@@ -374,6 +409,75 @@ impl Object {
 
         Ok(())
     }
+
+    fn parse_map(
+        &self,
+        section: &Section,
+        name: &str,
+        globals: &HashMap<&str, &[u8]>,
+    ) -> Result<Map, ParseError> {
+        let (def, data, kind) =
+            if name == ".bss" || name.starts_with(".data") || name.starts_with(".rodata") {
+                let mut data = section.data.to_vec();
+                let symbols: Vec<&Symbol> = self
+                    .symbols_by_index
+                    .iter()
+                    .filter(|(_, s)| {
+                        if let Some(idx) = s.section_index {
+                            idx == section.index
+                        } else {
+                            false
+                        }
+                    })
+                    .map(|(_, s)| s)
+                    .collect();
+                for symbol in symbols {
+                    if let Some(name) = &symbol.name {
+                        if let Some(symbol_data) = globals.get(name.as_str()) {
+                            if symbol_data.len() as u64 != symbol.size {
+                                return Err(ParseError::InvalidGlobalData {
+                                    name: name.to_string(),
+                                    sym_size: symbol.size,
+                                    data_size: symbol_data.len(),
+                                });
+                            }
+                            data.splice(
```

Comment:
You should check the range here as splice will panic otherwise

Reply:
fixed